quic/s2n-quic-transport/src/path/mod.rs

```diff
@@ -437,25 +437,11 @@ impl<Config: endpoint::Config> Path<Config> {
     //# PLPMTU.
     #[inline]
     pub fn clamp_mtu(&self, requested_size: usize, transmission_mode: transmission::Mode) -> usize {
-        let mtu = self.mtu(transmission_mode);
-
-        match self.state {
-            State::Validated => requested_size.min(mtu),
-
-            // https://github.com/aws/s2n-quic/issues/695
-            // Note: while a 3X check if performed, the `limit` value is not used
-            // to restrict the MTU. There are two reasons for this:
-            // - Expanding to the full MTU allows for MTU validation during connection migration.
-            // - Networking infrastructure cares more about number of packets than bytes for
-            // anti-amplification.
-            State::AmplificationLimited { tx_allowance } => {
-                if tx_allowance > 0 {
-                    requested_size.min(mtu)
-                } else {
-                    0
-                }
-            }
+        if self.at_amplification_limit() {
```

Comment:
This behavior always seemed a little weird to me, from what I can tell it is only used for the debug_assertion in [transmission.rs](https://github.com/aws/s2n-quic/blob/main/quic/s2n-quic-transport/src/connection/transmission.rs#L112-L115). 

Maybe move the debug assertion into this method, or just have `transmission.rs` do something like:
```rust
 debug_assert!(!self.context.path().is_amplification_limited(), "the amplification limit should be checked before trying to transmit");
```

Reply:
I'll just move the debug assertion into this method.

Reply:
https://github.com/aws/s2n-quic/pull/1494/commits/3c0766573de91c80cacf374bbca6d4eba3bdee3a

Reply:
looks like there are some unit tests that need to be fixed